src/compatTest/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPluginTests.kt

```diff
@@ -206,6 +206,102 @@ class NexusPublishPluginTests {
+                    }
+                }
+            }
+        """)
+
+        val otherStagingProfileId = "otherStagingProfileId"
+        val otherStagingRepositoryId = "orgexample-43"
+        stubStagingProfileRequest("/staging/profiles", mapOf("id" to STAGING_PROFILE_ID, "name" to "org.example"))
+        stubStagingProfileRequest("/staging/profiles", mapOf("id" to otherStagingProfileId, "name" to "org.example"), wireMockServer = otherServer)
+        stubCreateStagingRepoRequest("/staging/profiles/$STAGING_PROFILE_ID/start", STAGED_REPOSITORY_ID)
+        stubCreateStagingRepoRequest("/staging/profiles/$otherStagingProfileId/start", otherStagingRepositoryId, wireMockServer = otherServer)
+        expectArtifactUploads("/staging/deployByRepositoryId/$STAGED_REPOSITORY_ID")
+        expectArtifactUploads("/staging/deployByRepositoryId/$otherStagingRepositoryId", wireMockServer = otherServer)
+
+        // In order to test a potential race condition when running the publish tasks in parallel, we ask gradle to run the two get*StagingProfile
+        // tasks explicitly to demonstrate that if the output of each task is not segmented from each other by repository name the output of one
+        // get*StagingProfile task would overwrite the other and result in incorrect API requests getting sent to MyNexus repository
+        val result = run("getMyNexusStagingProfile", "getSomeOtherNexusStagingProfile", "publishToMyNexus", "publishToSomeOtherNexus")
+
+        assertSuccess(result, ":initializeMyNexusStagingRepository")
+        assertSuccess(result, ":initializeSomeOtherNexusStagingRepository")
+        assertThat(result.output).containsOnlyOnce("Created staging repository '$STAGED_REPOSITORY_ID' at ${server.baseUrl()}/repositories/$STAGED_REPOSITORY_ID/content/")
+        assertThat(result.output).containsOnlyOnce("Created staging repository '$otherStagingRepositoryId' at ${otherServer.baseUrl()}/repositories/$otherStagingRepositoryId/content/")
+        server.verify(postRequestedFor(urlEqualTo("/staging/profiles/$STAGING_PROFILE_ID/start"))
+            .withRequestBody(matchingJsonPath("\$.data[?(@.description == 'org.example:sample:0.0.1')]")))
+        otherServer.verify(postRequestedFor(urlEqualTo("/staging/profiles/$otherStagingProfileId/start"))
+            .withRequestBody(matchingJsonPath("\$.data[?(@.description == 'org.example:sample:0.0.1')]")))
+        assertUploadedToStagingRepo("/org/example/sample/0.0.1/sample-0.0.1.pom")
+        assertUploadedToStagingRepo("/org/example/sample/0.0.1/sample-0.0.1.jar")
+        assertUploadedToStagingRepo("/org/example/sample/0.0.1/sample-0.0.1.pom", stagingRepositoryId = otherStagingRepositoryId, wireMockServer = otherServer)
+        assertUploadedToStagingRepo("/org/example/sample/0.0.1/sample-0.0.1.jar", stagingRepositoryId = otherStagingRepositoryId, wireMockServer = otherServer)
+    }
+
+    @Test
+    fun `can get StagingProfileId from Nexus`() {
```

Comment:
We don't enforce a declared order of tests, nor skill other tests after the first failure, but for logic order in operations performed in read, I would put it as the second test in the class (right after `must be applied to root project`).

Reply:
Agreed! When I first wrote this test, I didn't realize I would be adding this into the chain of execution of the publish tasks 😃